xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1402,6 +1402,48 @@ namespace VIDEO
       if (!strTrailer.empty())
         movieDetails.m_strTrailer = strTrailer;
 
+      // Deal with 'Disc' subdirectories and add to/create a set
+      if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+              CSettings::SETTING_MYVIDEOS_HANDLEDISCN))
+      {
+        const size_t discPos = pItem->GetDiscPath(movieDetails.m_strFileNameAndPath);
+        if (discPos)
+        {
+          if (movieDetails.m_set.title.empty())
+          {
+            // Get existing sets
+            CFileItemList listItems;
+            const std::string baseDir = "videodb://movies/sets/?ignoreSingleMovieSets=false";
+            if (CDirectory::GetDirectory(baseDir, listItems, "", DIR_FLAG_DEFAULTS))
+            {
+              // See if set exists
+              for (const auto& item : listItems)
+              {
+                const std::string currentSet{
+                    m_database.GetSetById(item->GetVideoInfoTag()->m_iDbId)};
+                if (currentSet == movieDetails.m_strTitle)
+                {
+                  // Set exists
+                  movieDetails.SetSet(movieDetails.m_strTitle);
+                  break;
+                }
+              }
+              if (movieDetails.m_set.title.empty())
+              {
+                // Create set
+                const int idSet = m_database.AddSet(movieDetails.m_strTitle);
+                m_database.SetArtForItem(idSet, MediaTypeVideoCollection, art);
+                movieDetails.SetSet(movieDetails.m_strTitle);
+              }
+            }
+          }
+          // Add 'Disc n' to title
+          const size_t discEnd = movieDetails.m_strFileNameAndPath.find("/",discPos + 1);
+          movieDetails.m_strTitle +=
+              " (" + movieDetails.m_strFileNameAndPath.substr(discPos + 1, discEnd - discPos - 1) + ")";
```

Comment:
Shouldn't we localize "Disc" here and not take it from the file name?

And hard-coding "Disc n" surrounded by brackets is also not a good idea. In other languages other signs may be appropriate for what we usually put in brackets. So, we need a localizable string like "(Disc %s)" and use it here with a `StringUtils::Format` to fill in the number.

Reply:
I've changed it to use `StringUtils::Format` but I don't know how to localise the brackets.